cmd/thanos/store.go

```diff
@@ -140,7 +147,16 @@ func runStore(
 	filterConf *store.FilterConfig,
 	selectorRelabelConf *extflag.PathOrContent,
 	advertiseCompatibilityLabel bool,
+	consistencyDelay time.Duration,
 ) error {
+	consistencyDelayMetric := prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "thanos_consistency_delay_seconds",
+		Help: "Configured consistency delay in seconds.",
+	}, func() float64 {
+		return consistencyDelay.Seconds()
+	})
+	reg.MustRegister(consistencyDelayMetric)
```

Comment:
Since we have `NewConsistencyDelayMetaFilter` now perhaps the metrics could be registered there? We could pass the registry as: `extprom.WrapRegistererWithPrefix("thanos_", reg)` and then use a `component.Component` as the `Namespace`, and only then `consistency_delay_seconds` as the metric's name. I think it would be cleaner and less duplication between `cmd/thanos/store.go` and `cmd/thanos/compact.go`. What do you think?

Reply:
I agree with this, I have made the change